app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+            height: '100%',
+            alignItems: 'center',
+            justifyContent: 'center',
+            zIndex: 1000,
+        },
+        description: {
+            alignSelf: 'center',
+            marginBottom: 24,
+        },
+        text: {
+            ...typography('Body', 75),
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+    };
+});
+
+const FIELDS: { [id: string]: MessageDescriptor } = {
+    firstName: {
+        id: t('user.settings.general.firstName'),
+        defaultMessage: 'First Name',
+    },
+    lastName: {
+        id: t('user.settings.general.lastName'),
+        defaultMessage: 'Last Name',
+    },
+    username: {
+        id: t('user.settings.general.username'),
+        defaultMessage: 'Username',
+    },
+    nickname: {
+        id: t('user.settings.general.nickname'),
+        defaultMessage: 'Nickname',
+    },
+    position: {
+        id: t('user.settings.general.position'),
+        defaultMessage: 'Position',
+    },
+    email: {
+        id: t('user.settings.general.email'),
+        defaultMessage: 'Email',
+    },
+};
+
+const EditProfile = ({
+    componentId,
+    currentUser,
+    isModal,
+    isTablet,
+    lockedFirstName,
+    lockedLastName,
+    lockedNickname,
+    lockedPosition,
+}: EditProfileProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const keyboardAwareRef = useRef<KeyboardAwareScrollView>();
+    const firstNameRef = useRef<FloatingTextInputRef>(null);
+    const lastNameRef = useRef<FloatingTextInputRef>(null);
+    const usernameRef = useRef<FloatingTextInputRef>(null);
+    const emailRef = useRef<FloatingTextInputRef>(null);
+    const nicknameRef = useRef<FloatingTextInputRef>(null);
+    const positionRef = useRef<FloatingTextInputRef>(null);
+
+    const styles = getStyleSheet(theme);
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = useMemo(() => {
+        return isTablet ? null : {
+            id: 'update-profile',
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'edit_profile.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: buttonText,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    const closeButtonId = 'close-edit-profile';
```

Comment:
Nit: I wonder if we should move this constant out of the component.

Reply:
Done